doc/kci_testsuite.md

```diff
@@ -0,0 +1,300 @@
+
+3. Remember `script` path is relative to `kci_rootfs build --data-path` value. By default its `jenkins/debian/debos/script`
+   Here we need add new script named `buster-btrfs.sh`. Its a simple script to download the repo and build and install 
+   the btrfs-progs from source. 
+
+
+```
+$ cat jenkins/debian/debos/script/buster-btrfs.sh
+
+git clone https://github.com/kdave/btrfs-progs.git /btrfs-progs
+cd /btrfs-progs
+./autogen.sh && ./configure
+make
+make install
+```
+
+
+4. `test_overlay` option is used to create specific directory layout inside the rootfs image. In our case, 
+
+```
+$ tree jenkins/debian/debos/overlays/btrfs/
+jenkins/debian/debos/overlays/btrfs/
+└── usr
+    └── bin
+        └── btrfs-verify-cli.sh
+```
+
+   `/usr/bin/btrfs-verify-cli.sh` will be available on `buster-btrfs` rootfs image. These overlay scripts normally ran from LAVA job.
+
+```
+$ cat jenkins/debian/debos/overlays/btrfs/btrfs-verify-cli.sh
+
+cd  /btrfs-progs && TEST=001\* ./tests/cli-tests.sh
+```
+
+5. Make sure execute permission is applied on the scripts. 
+
+```
+chmod +x jenkins/debian/debos/overlays/btrfs/usr/bin/btrfs-verify-cli.sh
+chmod +x jenkins/debian/debos/script/buster-btrfs.sh
+```
+
+6. Now build `buster-btrfs` rootfs image using kci_rootfs. 
+
+7. Once rootfs image is created, you can use it with appropriate kernel image. Make sure that kernel config has 
+   appropriate btrfs configs and other file system specific entries enabled. 
+
+8. Add appropriate test entry with LAVA job definition for `btrfs-verify-cli.sh`. For example,
+
+```
+run:
+    steps:
+        - lava-test-case test-btrfs-cli-001 --shell ls /usr/bin/btrfs-verify-cli.sh
+
+```
+
+If you like read more about LAVA job definition, please check [here](https://docs.lavasoftware.org/lava/writing-tests.html)
+
+
+### How to add tests to LAVA
+
+#### Prerequisites
+
+- Rootfs image is built 
+- Rootfs image boots on a platform
+- Tests can be run
+
+If you're facing issues with tha above you may want to go through the section "How to add test suite to a rootfs image"
+
+#### Adding new test plan to LAVA
+
+1. Use the rootfs image to boot the device and produce a LAVA job definition for the tests.
+Upload rootfs image to kernelci-backend (storage)
+
+```
+kci_rootfs upload --rootfs-dir /path/to/rootf/image/ --upload-path server/path --api http://api.kernelci --token KCI_API_TOKEN
+```
+
+Add rootfs definition to `test-configs.yaml`
+
+```yaml
+  debian_buster-v4l2_ramdisk:
+    type: debian
+    ramdisk: 'buster-v4l2/20200626.0/{arch}/rootfs.cpio.gz'
+```
+
+`ramdisk` path must be relative to storage server URL.
+
+Detailed description of the test configuration can be found [here](https://github.com/kernelci/kernelci-doc/wiki/Test-configurations)
+
+2. Add LAVA job definition template
+
+Create appropriate directory structure under `templates` directory.
+
+```
+mkdir templates/v4l2-compliance/
```

Comment:
Again, maybe use the btrfs test as per the example in the first part of this page?